pyvisa/resources/messagebased.py

```diff
@@ -469,17 +469,44 @@ def read_binary_values(self, datatype='f', is_big_endian=False,
         :rtype: type(container)
 
         """
-        block = self._read_raw()
+        block = b''
         expected_length = 0
 
         if header_fmt == 'ieee':
-            offset, data_length = util.parse_ieee_block_header(block)
+            hash = self.read_bytes(1)                   # Hash '#'
+            digits = self.read_bytes(1)                 # Digits
```

Comment:
Reading multiple times is not free, we should group those readings as much as possible. This was actually the motivation behind the current code: avoid multiple reading when possible by reading a large chunk. However it this causes issues I am fine changing it but we should try to limit the overhead.

Reply:
I can combine those 2 into 1 call